src/textual/color.py

```diff
@@ -385,6 +385,53 @@ def blend(
             new_alpha,
         )
 
+    @lru_cache(maxsize=1024)
+    def hsl_blend(
+        self, destination: Color, factor: float, alpha: float | None = None
+    ) -> Color:
+        """Generate a new color between two colors in the HSL color space.
+
+        This method calculates a new color on a gradient using the HSL color space.
+        The position on the gradient is given by `factor`, which is a float between -1 and 1, where 0 is the original color, and 1 or -1 is the `destination` color.
+        The sign of `factor` affects the direction of the hue angle, a positive number is in the clockwise direction, a negative number is in the counter-clockwise direction.
+        For lightness and saturation, only the absolute value of `factor` is used.
+        A value of `factor` between the two extremes produces a color somewhere between the two end points.
+
+        Args:
+            destination: Another color.
+            factor: A blend factor, -1 -> 1.
```

Comment:
Nit-pick: I'd spell out that it's "between -1 and 1", and perhaps make clear if it's exclusive or inclusive.